core/arch/arm/pta/device.c

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2019, Linaro Limited
+ */
+
+/*
+ * This pseudo TA is used by normal world OS TEE driver to fetch pseudo TA's
+ * UUIDs which can act as TEE bus devices.
+ */
+
+#include <kernel/pseudo_ta.h>
+#include <kernel/tee_ta_manager.h>
+#include <tee/uuid.h>
+#include <pta_device.h>
+#include <string.h>
+#include <user_ta_header.h>
+
+#define PTA_NAME "device.pta"
+
+static TEE_Result get_devices(uint32_t types,
+			      TEE_Param params[TEE_NUM_PARAMS])
+{
+	const struct pseudo_ta_head *ta;
+	TEE_UUID *device_uuid = NULL;
+	uint8_t uuid_octet[sizeof(TEE_UUID)];
+	size_t ip_size, op_size = 0;
+	TEE_Result res = TEE_SUCCESS;
+
+	if (types != TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_OUTPUT,
+				     TEE_PARAM_TYPE_NONE,
+				     TEE_PARAM_TYPE_NONE,
+				     TEE_PARAM_TYPE_NONE)) {
+		EMSG("bad parameters types: 0x%" PRIx32, types);
```

Comment:
I don't think this `EMSG()` is useful, the return code should be good enough (error messages do take space in the TEE binary...).
Then you can drop the braces.

Reply:
ok